lib/convenience_functions.rb

```diff
@@ -63,13 +63,14 @@ def self.load_symbolized_json
   end
 
   def self.read_filelist(path, always_calcuate_from_disk: false)
-    filelist = File.readlines(path, chomp: true)
+    # The filelist should not include commented lines since it will always add to the number of files.
+    filelist = Pathname.new(path).readlines.reject { |line| line.start_with?('#') }.map { |line| line.chomp }
 
-    if filelist.first&.start_with?('# Total size') && !always_calcuate_from_disk
-      total_size, *contents = filelist
-      return [total_size[/Total size: (\d+)/, 1].to_i, contents]
+    firstline = File.open(path, &:gets).chomp.to_s
+    if firstline.start_with?('# Total size') && !always_calcuate_from_disk
+      return [firstline[/Total size: (\d+)/, 1].to_i, filelist, filelist.count]
     else
-      return [get_package_disk_size(filelist), filelist]
+      return [get_package_disk_size(filelist), filelist, filelist.count]
     end
   end
```

Comment:
What about just returning filelist.count - 1 since there will only be one commented line?